src/lib/stitching/lib/getTypesFromSchema.ts

```diff
@@ -52,7 +52,35 @@ export const getRootFieldsFromSchema = async (schema: GraphQLSchema) => {
   }
 
   // Return the real data
-  return response.data!.__schema.queryType.fields.map(t => t.name)
+  return response.data!.__schema.queryType.fields.map(t => t.name) as string[]
+}
+
+/**
+ * An introspection query that pulls out the fields for a type you provide
+ */
+export const getFieldsForTypeFromSchema = async (
+  type: string,
+  schema: GraphQLSchema
+) => {
+  const q = gql`
+    query {
+      __type(name: "${type}") {
+        fields {
+          name
+        }
+      }
+    }
+  `
+  const response = await graphql(schema, q, null, null)
+  // Throw an error to make it easy to read in the failing test
+  if (response.errors) {
+    throw new Error(
+      `Got errors from GQL: ${JSON.stringify(response.errors, null, "  ")}`
+    )
+  }
+
+  // Return the real data
+  return response.data!.__type.fields.map(t => t.name) as string[]
```

Comment:
Remind me, why do we perform an introspection query like this over using the graphql-js API and operating on the schema object directly?

Reply:
This felt simpler (+ very high level) mainly - and provided a pattern for folks wanting to do other test-y queries like this, but I'm open to using a higher level API for this also!

Reply:
That said, I think the implementations can change anytime to use those APIs and we just have a high level set of functions